src/test/java/org/opentripplanner/apis/gtfs/PatternByServiceDatesFilterTest.java

```diff
@@ -0,0 +1,159 @@
+    return List.of(
+      Arguments.of(null, null),
+      Arguments.of(parse("2024-05-02"), parse("2024-05-01")),
+      Arguments.of(parse("2024-05-03"), parse("2024-05-01"))
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("invalidRangeCases")
+  void invalidRange(LocalDate start, LocalDate end) {
+    assertThrows(
+      IllegalArgumentException.class,
+      () ->
+        new PatternByServiceDatesFilter(
+          new LocalDateRange(start, end),
+          r -> List.of(),
+          d -> List.of()
+        )
+    );
+  }
+
+  static List<Arguments> validRangeCases() {
+    return List.of(
+      Arguments.of(parse("2024-05-02"), parse("2024-05-02")),
+      Arguments.of(parse("2024-05-02"), parse("2024-05-03")),
+      Arguments.of(null, parse("2024-05-03")),
+      Arguments.of(parse("2024-05-03"), null)
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("validRangeCases")
+  void validRange(LocalDate start, LocalDate end) {
+    assertDoesNotThrow(() ->
+      new PatternByServiceDatesFilter(
+        new LocalDateRange(start, end),
+        r -> List.of(),
+        d -> List.of()
+      )
+    );
+  }
+
+  static List<Arguments> ranges() {
+    return List.of(
+      Arguments.of(null, parse("2024-05-03"), NOT_REMOVED),
+      Arguments.of(parse("2024-05-03"), null, NOT_REMOVED),
+      Arguments.of(parse("2024-05-01"), null, NOT_REMOVED),
+      Arguments.of(null, parse("2024-04-30"), REMOVED),
+      Arguments.of(null, parse("2024-05-01"), REMOVED),
+      Arguments.of(parse("2024-05-02"), parse("2024-05-02"), REMOVED),
+      Arguments.of(parse("2024-05-02"), parse("2024-05-03"), REMOVED),
+      Arguments.of(parse("2024-05-02"), parse("2024-06-01"), REMOVED),
+      Arguments.of(parse("2025-01-01"), null, REMOVED),
+      Arguments.of(parse("2025-01-01"), parse("2025-01-02"), REMOVED),
+      Arguments.of(null, parse("2023-12-31"), REMOVED),
+      Arguments.of(parse("2023-12-31"), parse("2024-04-30"), REMOVED)
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("ranges")
+  void filterPatterns(LocalDate start, LocalDate end, FilterExpectation expectation) {
+    var filter = mockFilter(start, end);
+
+    var filterInput = List.of(PATTERN_1);
+    var filterOutput = filter.filterPatterns(filterInput);
+
+    if (expectation == NOT_REMOVED) {
+      assertEquals(filterOutput, filterInput);
+    } else {
+      assertEquals(List.of(), filterOutput);
+    }
+  }
+
+  @ParameterizedTest
+  @MethodSource("ranges")
+  void filterRoutes(LocalDate start, LocalDate end, FilterExpectation expectation) {
+    var filter = mockFilter(start, end);
+
+    var filterInput = List.of(ROUTE_1);
+    var filterOutput = filter.filterRoutes(filterInput.stream());
+
+    if (expectation == NOT_REMOVED) {
+      assertEquals(filterOutput, filterInput);
+    } else {
+      assertEquals(List.of(), filterOutput);
+    }
+  }
+
+  private static PatternByServiceDatesFilter mockFilter(LocalDate start, LocalDate end) {
```

Comment:
This is not mocking the filter, rename to `createFilter`.

Reply:
https://github.com/opentripplanner/OpenTripPlanner/pull/5869/commits/b6de8304340acb952d88a46cd9589f569d3c7db9